app/assets/javascripts/discourse/app/initializers/animated-images.js

```diff
@@ -8,29 +10,42 @@ export default {
   initialize() {
     withPluginApi("0.8.7", (api) => {
       function _cleanUp() {
-        Object.values(_gifClickHandlers || {}).forEach((handler) =>
-          handler.removeEventListener("click", _handleClick)
-        );
+        Object.values(_gifClickHandlers || {}).forEach((handler) => {
+          handler.removeEventListener("click", _handleEvent);
+          handler.removeEventListener("load", _handleEvent);
+        });
 
         _gifClickHandlers = {};
       }
 
-      function _handleClick(event) {
+      function _handleEvent(event) {
         const img = event.target;
         if (img && !img.previousSibling) {
-          let canvas = document.createElement("canvas");
-          canvas.width = img.width;
-          canvas.height = img.height;
-          canvas.getContext("2d").drawImage(img, 0, 0, img.width, img.height);
-          canvas.setAttribute("aria-hidden", "true");
-          canvas.setAttribute("role", "presentation");
-
-          img.parentNode.classList.add("paused-animated-image");
-          img.parentNode.insertBefore(canvas, img);
+          _pauseAnimation(img, { manualPause: true });
         } else {
-          img.previousSibling.remove();
-          img.parentNode.classList.remove("paused-animated-image");
+          _resumeAnimation(img);
+        }
+      }
+
+      function _pauseAnimation(img, opts = {}) {
+        let canvas = document.createElement("canvas");
+        canvas.width = img.width;
+        canvas.height = img.height;
+        canvas.getContext("2d").drawImage(img, 0, 0, img.width, img.height);
+        canvas.setAttribute("aria-hidden", "true");
+        canvas.setAttribute("role", "presentation");
+
+        if (opts.manualPause) {
+          img.classList.add("manually-paused");
         }
+        img.parentNode.classList.add("paused-animated-image");
+        img.parentNode.insertBefore(canvas, img);
+      }
+
+      function _resumeAnimation(img) {
+        img.previousSibling.remove();
+        img.parentNode.classList.remove("paused-animated-image");
```

Comment:
Just a minor note, know that you can add/remove multiple classes at once:

```
div.classList.remove("foo", "bar", "baz");
```

https://developer.mozilla.org/en-US/docs/Web/API/Element/classList